### PR TITLE
Users api cars refactoring

### PR DESCRIPTION
**Pull request by BorodaUA**
Hi team, please review changes that I made in formal "/cars" endpoint, now it is nested with "/drivers/{id}/cars".
-Renamed driver_id > driver, model field.
-Added  serializer and authentication classes
-CarSerializer break up into read and write serializers
-Added nested routing to make /drivers/{id}/cars ,endpoint.

### CODE REVIEW

File: usefull_app/user_api/serializers.py [hunk 1]
```diff
@@ -86,20 +87,56 @@ def update(self, instance, validated_data):
...
+
+class CarWriteSerializer(serializers.ModelSerializer):
+    """
+    Car model WRITE serializer for "/cars" endpoint.
+    """
+
+    class Meta:
+        model = Car
+        fields = [
+            # 'id',
```
Comment (tropicoo) [inline usefull_app/user_api/serializers.py:114]: Did you forget to remove this line?

File: usefull_app/user_api/serializers.py [hunk 2]
```diff
@@ -147,15 +184,15 @@ class TripSerializer(serializers.ModelSerializer):
     route = RouteSerializer(source='route_id')
     passengers = PassengerSerializer(many=True, source='passenger_id')
     driver = DriverSerializer(source='driver_id')
-    car = CarSerializer(source='car_id')
+    # car = CarSerializer(source='car_id')
```
Comment (tropicoo) [inline usefull_app/user_api/serializers.py:187]: Same here

File: usefull_app/user_api/serializers.py [hunk 3]
```diff
@@ -147,15 +184,15 @@ class TripSerializer(serializers.ModelSerializer):
...
 
     class Meta:
         model = Trip
         fields = [
             'id',
             'route',
             'driver',
-            'car',
+            # 'car',
```
Comment (tropicoo) [inline usefull_app/user_api/serializers.py:195]: And here

**Review by tropicoo — CHANGES_REQUESTED** [review]

File: usefull_app/user_api/serializers.py (hunk 1 above)
Comment (BorodaUA) [reply usefull_app/user_api/serializers.py]: yes, fixed in new commit

File: usefull_app/user_api/serializers.py (hunk 2 above)
Comment (BorodaUA) [reply usefull_app/user_api/serializers.py]: yes, fixed in new commit.

File: usefull_app/user_api/serializers.py (hunk 3 above)
Comment (BorodaUA) [reply usefull_app/user_api/serializers.py]: and here also

**Review by tropicoo — APPROVED** [review]

**Pull request merged**